get-started/part2.md

```diff
@@ -136,149 +140,130 @@ def hello():
 
 if __name__ == "__main__":
     app.run(host='0.0.0.0', port=80)
 ```
 
-Now we see that `pip install -r requirements.txt` installs the Flask and Redis
-libraries for Python, and the app prints the environment variable `NAME`, as
-well as the output of a call to `socket.gethostname()`. Finally, because Redis
-isn't running (as we've only installed the Python library, and not Redis
-itself), we should expect that the attempt to use it here will fail and produce
-the error message.
+Scroll up and take another look at `Dockerfile`. The `ADD` command bundles
+`app.py` and `requirements.txt` into the build. The `EXPOSE` command makes the
+output from `app.py` accessible over HTTP.
 
-> **Note**: Accessing the name of the host when inside a container retrieves the
-container ID, which is like the process ID for a running executable.
+The command, `RUN pip install -r requirements.txt`, installs the Python
+libraries for Flask and Redis. The app prints the environment variable `NAME`
+and the output of the call to `socket.gethostname()`, _which is the container ID._
 
-That's it! You don't need Python or anything in `requirements.txt` on your
-system, nor will building or running this image install them on your system. It
-doesn't seem like you've really set up an environment with Python and Flask, but
-you have.
+### Build the app
 
-## Build the app
-
-We are ready to build the app. Make sure you are still at the top level of your new directory. Here's what `ls` should show:
+We are ready to build the app. Make sure you are still at the top level of your
+new directory and check your files:
 
 ```shell
 $ ls
 Dockerfile		app.py			requirements.txt
 ```
 
-Now run the build command. This creates a Docker image, which we're going to
-tag using `-t` so it has a friendly name.
+Build the image with the files in _this_ directory (`.`) and tag it with a
+meaningful name such as `hellomoby`:
 
 ```shell
-docker build -t friendlyhello .
+$ docker build ---tag hellomoby .
 ```
 
-Where is your built image? It's in your machine's local Docker image registry:
+You can find `hellomoby` in your machine's local Docker image registry:
 
 ```shell
-$ docker images
-
-REPOSITORY            TAG                 IMAGE ID
-friendlyhello         latest              326387cea398
+$ docker image ls
+REPOSITORY      TAG          IMAGE ID           CREATED              SIZE
+hellomoby       latest       88b087bf9a5e       2 minutes ago        148MB
+python          2.7-slim     4fd30fc83117       3 weeks ago          138MB
+hello-world     latest       f2a91732366c       6 weeks ago          1.85kB
+...
 ```
 
-## Run the app
+### Run the app
 
-Run the app, mapping your machine's port 4000 to the container's published port
-80 using `-p`:
+Run the app, and use the port flag, `-p`, to map port 4000 on your machine to
+port 80 published by the container:
 
 ```shell
-docker run -p 4000:80 friendlyhello
+$ docker run -p 4000:80 hellomoby
```

Comment:
Can you use the new port-mapping syntax? I went through and updated most of the relevant examples to use it a while ago, like here: https://docs.docker.com/engine/swarm/stack-deploy/#set-up-a-docker-registry

Reply:
I'm not in swarm mode yet so ...